### ISSUE TITLE
`kedro-mlflow` doesn't work with `OmegaConfigLoader`

### ISSUE THREAD

**Issue by ankatiyar** [Galileo-Galilei/kedro-mlflow#430]
**_If you like the repo, please give it a :star:_**

## Description
When trying to do a `kedro run` with the `kedro-mlflow` plugin installed on a project that uses `OmegaConfigLoader` it crashes.

## Context
`OmegaConfigLoader` does not have a `get()` method which is used here - 
https://github.com/Galileo-Galilei/kedro-mlflow/blob/845ad919c9dbd020e948e8adc2e0f9064de1ef68/kedro_mlflow/framework/hooks/mlflow_hook.py#L62-L63

After these changes in the Kedro repo - https://github.com/kedro-org/kedro/pull/1966 & https://github.com/kedro-org/kedro/pull/1870

Ideally the patterns should be set in `settings.py`:
```
CONFIG_LOADER_ARGS = {
      "config_patterns": {
          "mlflow" : ["mlflow*", "mlflow*/**"],
      }
}
```
And then `mlflow_hook.py` can access it like `context.config_loader["mlflow"]`

## Steps to Reproduce
Install `kedro-mlflow`
1. `kedro new --starter=spaceflights`
2. in `settings.py` -
```
from kedro.config import OmegaConfigLoader
CONFIG_LOADER_CLASS = OmegaConfigLoader
```
3. `kedro run`

## Actual Result
Traceback :

```
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/bin/kedro:8 in <module>                            │
│                                                                                                  │
│ /Users/ankita_katiyar/kedro/kedro/kedro/framework/cli/cli.py:211 in main                         │
│                                                                                                  │
│   208 │   """                                                                                    │
│   209 │   _init_plugins()                                                                        │
│   210 │   cli_collection = KedroCLI(project_path=Path.cwd())                                     │
│ ❱ 211 │   cli_collection()                                                                       │
│   212                                                                                            │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/click/core.py:1130 in │
│ __call__                                                                                         │
│                                                                                                  │
│ /Users/ankita_katiyar/kedro/kedro/kedro/framework/cli/cli.py:139 in main                         │
│                                                                                                  │
│   136 │   │   )                                                                                  │
│   137 │   │                                                                                      │
│   138 │   │   try:                                                                               │
│ ❱ 139 │   │   │   super().main(                                                                  │
│   140 │   │   │   │   args=args,                                                                 │
│   141 │   │   │   │   prog_name=prog_name,                                                       │
│   142 │   │   │   │   complete_var=complete_var,                                                 │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/click/core.py:1055 in │
│ main                                                                                             │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/click/core.py:1657 in │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/click/core.py:1404 in │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/click/core.py:760 in  │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/ankita_katiyar/kedro/kedro/kedro/framework/cli/project.py:459 in run                      │
│                                                                                                  │
│   456 │   with KedroSession.create(                                                              │
│   457 │   │   env=env, conf_source=conf_source, extra_params=params                              │
│   458 │   ) as session:                                                                          │
│ ❱ 459 │   │   session.run(                                                                       │
│   460 │   │   │   tags=tag,                                                                      │
│   461 │   │   │   runner=runner(is_async=is_async),                                              │
│   462 │   │   │   node_names=node_names,                                                         │
│                                                                                                  │
│ /Users/ankita_katiyar/kedro/kedro/kedro/framework/session/session.py:371 in run                  │
│                                                                                                  │
│   368 │   │   session_id = self.store["session_id"]                                              │
│   369 │   │   save_version = session_id                                                          │
│   370 │   │   extra_params = self.store.get("extra_params") or {}                                │
│ ❱ 371 │   │   context = self.load_context()                                                      │
│   372 │   │                                                                                      │
│   373 │   │   name = pipeline_name or "__default__"                                              │
│   374                                                                                            │
│                                                                                                  │
│ /Users/ankita_katiyar/kedro/kedro/kedro/framework/session/session.py:275 in load_context         │
│                                                                                                  │
│   272 │   │   │   extra_params=extra_params,                                                     │
│   273 │   │   │   hook_manager=self._hook_manager,                                               │
│   274 │   │   )                                                                                  │
│ ❱ 275 │   │   self._hook_manager.hook.after_context_created(  # pylint: disable=no-member        │
│   276 │   │   │   context=context                                                                │
│   277 │   │   )                                                                                  │
│   278                                                                                            │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_hooks.py:265  │
│ in __call__                                                                                      │
│                                                                                                  │
│   262 │   │   else:                                                                              │
│   263 │   │   │   firstresult = False                                                            │
│   264 │   │                                                                                      │
│ ❱ 265 │   │   return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)        │
│   266 │                                                                                          │
│   267 │   def call_historic(self, result_callback=None, kwargs=None):                            │
│   268 │   │   """Call the hook with given ``kwargs`` for all registered plugins and              │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_manager.py:80 │
│ in _hookexec                                                                                     │
│                                                                                                  │
│    77 │   def _hookexec(self, hook_name, methods, kwargs, firstresult):                          │
│    78 │   │   # called from all hookcaller instances.                                            │
│    79 │   │   # enable_tracing will set its own wrapping function at self._inner_hookexec        │
│ ❱  80 │   │   return self._inner_hookexec(hook_name, methods, kwargs, firstresult)               │
│    81 │                                                                                          │
│    82 │   def register(self, plugin, name=None):                                                 │
│    83 │   │   """Register a plugin and return its canonical name or ``None`` if the name         │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_manager.py:32 │
│ 7 in traced_hookexec                                                                             │
│                                                                                                  │
│   324 │   │   │   │   lambda: oldcall(hook_name, hook_impls, kwargs, firstresult)                │
│   325 │   │   │   )                                                                              │
│   326 │   │   │   after(outcome, hook_name, hook_impls, kwargs)                                  │
│ ❱ 327 │   │   │   return outcome.get_result()                                                    │
│   328 │   │                                                                                      │
│   329 │   │   self._inner_hookexec = traced_hookexec                                             │
│   330                                                                                            │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_result.py:60  │
│ in get_result                                                                                    │
│                                                                                                  │
│   57 │   │   │   return self._result                                                             │
│   58 │   │   else:                                                                               │
│   59 │   │   │   ex = self._excinfo                                                              │
│ ❱ 60 │   │   │   raise ex[1].with_traceback(ex[2])                                               │
│   61                                                                                             │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_result.py:33  │
│ in from_call                                                                                     │
│                                                                                                  │
│   30 │   │   __tracebackhide__ = True                                                            │
│   31 │   │   result = excinfo = None                                                             │
│   32 │   │   try:                                                                                │
│ ❱ 33 │   │   │   result = func()                                                                 │
│   34 │   │   except BaseException:                                                               │
│   35 │   │   │   excinfo = sys.exc_info()                                                        │
│   36                                                                                             │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_manager.py:32 │
│ 4 in <lambda>                                                                                    │
│                                                                                                  │
│   321 │   │   def traced_hookexec(hook_name, hook_impls, kwargs, firstresult):                   │
│   322 │   │   │   before(hook_name, hook_impls, kwargs)                                          │
│   323 │   │   │   outcome = _Result.from_call(                                                   │
│ ❱ 324 │   │   │   │   lambda: oldcall(hook_name, hook_impls, kwargs, firstresult)                │
│   325 │   │   │   )                                                                              │
│   326 │   │   │   after(outcome, hook_name, hook_impls, kwargs)                                  │
│   327 │   │   │   return outcome.get_result()                                                    │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_callers.py:60 │
│ in _multicall                                                                                    │
│                                                                                                  │
│   57 │   │   │   except StopIteration:                                                           │
│   58 │   │   │   │   pass                                                                        │
│   59 │   │                                                                                       │
│ ❱ 60 │   │   return outcome.get_result()                                                         │
│   61                                                                                             │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_result.py:60  │
│ in get_result                                                                                    │
│                                                                                                  │
│   57 │   │   │   return self._result                                                             │
│   58 │   │   else:                                                                               │
│   59 │   │   │   ex = self._excinfo                                                              │
│ ❱ 60 │   │   │   raise ex[1].with_traceback(ex[2])                                               │
│   61                                                                                             │
│                                                                                                  │
│ /Users/ankita_katiyar/opt/anaconda3/envs/new2/lib/python3.10/site-packages/pluggy/_callers.py:39 │
│ in _multicall                                                                                    │
│                                                                                                  │
│   36 │   │   │   │   │   except StopIteration:                                                   │
│   37 │   │   │   │   │   │   _raise_wrapfail(gen, "did not yield")                               │
│   38 │   │   │   │   else:                                                                       │
│ ❱ 39 │   │   │   │   │   res = hook_impl.function(*args)                                         │
│   40 │   │   │   │   │   if res is not None:                                                     │
│   41 │   │   │   │   │   │   results.append(res)                                                 │
│   42 │   │   │   │   │   │   if firstresult:  # halt further impl calls                          │
│                                                                                                  │
│ /Users/ankita_katiyar/kedro_projects/kedro-mlflow/kedro_mlflow/framework/hooks/mlflow_hook.py:72 │
│ in after_context_created                                                                         │
│                                                                                                  │
│    69 │   │   │   # is commented out. In this situation there is no MissingConfigException       │
│    70 │   │   │   # but we got an empty dict                                                     │
│    71 │   │   │   conf_mlflow_yml = {}                                                           │
│ ❱  72 │   │   mlflow_config = KedroMlflowConfig.parse_obj({**conf_mlflow_yml})                   │
│    73 │   │                                                                                      │
│    74 │   │   self._already_active_mlflow = False                                                │
│    75 │   │   if mlflow.active_run():                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: 'str' object is not a mapping
```

## Your Environment

* `kedro` : 0.18.11
*  `kedro-mlflow` : 0.11.8
* Python version used (`python -V`): 3.10
* Operating system and version: mac

## Does the bug also happen with the last version on master?
Yes

**Comment by Galileo-Galilei** [comment]
Thank you very much @ankatiyar for the bug report, and for pointing out a solution. I'll commit a fix soon, I just need to figure out how to deal with retrocompatibilty with other ``ConfigLoader``.

**Comment by marrrcin** [comment]
@Galileo-Galilei https://github.com/getindata/kedro-azureml/blob/a040b3c65c57a38cf6a64ca0d0792e471abe6911/kedro_azureml/manager.py#L43 :)

**Comment by ankatiyar** [comment]
I believe the dict like syntax, `context.config_loader['mlflow']` should work for all config loaders.

**Comment by marrrcin** [comment]
But the dict syntax will work only if you have 
```
CONFIG_LOADER_ARGS = {
      "config_patterns": {
          "mlflow" : ["mlflow*", "mlflow*/**"],
      }
}
```

The `.get()` syntax from the legacy `ConfigLoader` works without it.

**Comment by Galileo-Galilei** [comment]
Hi @ankatiyar, this is fixed in ``kedro-mlflow==0.11.9`` which was just released on PyPI, thanks for reporting the bug!
